utils/driver_options/set_affinity_options.py

```diff
@@ -41,7 +41,24 @@ def __init__(self):
         self._mainaffinity_to_driver_identity = raw_config["mainaffinity_to_driver_identity"]
         self._txaffinity_to_driver_identity = raw_config["txaffinity_to_driver_identity"]
         self._rxaffinity_to_driver_identity = raw_config["rxaffinity_to_driver_identity"]
-        self._device_str = raw_config["devices"]
+        self._device_str = raw_config["device_options"]
+
+        # Parse N200 array and construct devices string
+        devices_map = {}
+        for n200 in raw_config["n200s"]:
+            rx = bool(n200["rx"])
+            tx = bool(n200["tx"])
+            rx_int = bool(n200["rx_int"])
+
+            if rx or tx or rx_int:
+                device_num = int(n200["main_antenna"])
+                devices_map[device_num] = n200["addr"]
+
+        addr_idx = 0
+        for device_num in sorted(devices_map.keys()):
```

Comment:
I think its more pythonic and future-proof to use the python built-in enumerate(...) here instead of keeping track of your own addr_idx